### PR TITLE
Dan/expose disconnected status in svc groups

### PR DESCRIPTION
**Pull request by danielsdeleo**
### :nut_and_bolt: Description: What code changed, and why?

Includes information about disconnected services in the APIs for services and service groups:
* Service groups now includes a `disconnected_count` field to show the number of services that have their disconnected flag set to true.
* Services have a boolean field for `disconnected`, a `last_event_occurred_at` field with a UTC timestamp for the time the last event was received, and `last_event_since` which gives an English "time since" description of the elapsed time since the last event (e.g., `2 minutes 57 seconds`).

### :chains: Related Resources

#1184

### :athletic_shoe: How to Build and Test the Change

Build applications-service and automate-gateway; start all services.

To make services appear as disconnected more quickly, you will likely want to:

```
export T=$(get_admin_token)
curl -k -H "api-token: $T" -X POST https://a2-dev.test/apis/beta/retention/service_groups/disconnected_services/config -d '{"running":true,"threshold":"2m"}'
```

You can run `applications_populate_database` and wait 2-3 minutes for the services to become disconnected.

The new fields described above will then be seen in the API responses at URL paths like `/apis/beta/applications/service-groups` and `/apis/beta/applications/service-groups/:sgId`

### :white_check_mark: Checklist

- [x] Tests added/updated?

### CODE REVIEW

File: components/applications-service/pkg/storage/postgres/service_groups.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ const (
       ,s.application as app_name
       ,s.environment as environment
       ,s.service_group_name_suffix as name_suffix
+			,COUNT(s.disconnected) FILTER (WHERE s.disconnected = 'true') AS disconnected_count
```
Comment (srenatus) [inline components/applications-service/pkg/storage/postgres/service_groups.go]: Is `disconnected` a `text`? If it's bool, we should be able to do `FILTER (WHERE s.disconnected)`

File: components/applications-service/integration_test/services_test.go [hunk 2]
```diff
@@ -162,6 +173,49 @@ func TestGetServicesMultiService(t *testing.T) {
...
+
+	assert.Equal(t, 6, len(res.Services))
+	for _, s := range res.Services {
+		assert.True(t, s.Disconnected, "expected the service to be marked as disconnected")
+		expectedTime, err := ptypes.TimestampProto(eventTime)
+		require.NoError(t, err)
+		assert.Equal(t, expectedTime, s.LastEventOccurredAt)
+		// Ideally we could do:
+		//   assert.Equal(t, "30 minutes", s.LastEventSince)
+		// but we _usually_ get something like "29 minutes, 59 seconds" because we
```
Comment (srenatus) [inline components/applications-service/integration_test/services_test.go]: Would [`WithinDuration`](https://godoc.org/github.com/stretchr/testify/assert#WithinDuration) help?

Comment (danielsdeleo) [reply components/applications-service/integration_test/services_test.go]: Not quite, we have code within the API that makes a string (kinda like a similar helper function in ruby on rails) with the relative time in words, I'm checking the content of that string.

File: components/applications-service/pkg/storage/postgres/service_groups.go (hunk 1 above)
Comment (danielsdeleo) [reply components/applications-service/pkg/storage/postgres/service_groups.go]: I'll give that a go, thanks!

File: components/applications-service/integration_test/services_test.go (hunk 2 above)
Comment (srenatus) [reply components/applications-service/integration_test/services_test.go]: Ah now I get it. I guess you could parse the duration string in the test... But probably not worth it. ;)

**Pull request merged**